examples/demo/src/App.js

```diff
@@ -30,61 +30,61 @@ const i18nProvider = polyglotI18nProvider(locale => {
     return englishMessages;
 }, 'en');
 
-class App extends Component {
-    state = { dataProvider: null };
+const App = () => {
+    const [dataProvider, setDataProvider] = useState(null);
 
-    async componentDidMount() {
-        this.restoreFetch = await fakeServerFactory(
-            process.env.REACT_APP_DATA_PROVIDER
-        );
-
-        const dataProvider = await dataProviderFactory(
-            process.env.REACT_APP_DATA_PROVIDER
-        );
+    useEffect(() => {
+        let restoreFetch;
 
-        this.setState({ dataProvider });
-    }
+        const fetchDataProvider = async () => {
+            restoreFetch = await fakeServerFactory(
+                process.env.REACT_APP_DATA_PROVIDER
+            );
 
-    componentWillUnmount() {
-        this.restoreFetch();
-    }
+            setDataProvider(
+                await dataProviderFactory(process.env.REACT_APP_DATA_PROVIDER)
+            );
+        };
 
-    render() {
-        const { dataProvider } = this.state;
+        fetchDataProvider();
 
-        if (!dataProvider) {
-            return (
-                <div className="loader-container">
```

Comment:
I prefer the same loader for the data Generator waiting time and the data provider waiting time, so please don't use react-admin's Loading component in the demo.

Reply:
Ok, done.